nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1987 @@
+				},
+				"transport_zone_profile_id": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					Description: "Identifiers of the transport zone profiles associated with this transport zone endpoint on this transport node",
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+func getUplinksSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Optional:    true,
+		Description: "Uplink/LAG of VMware vSphere Distributed Switch connected to the HostSwitch",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"uplink_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Uplink name from UplinkHostSwitch profile",
+				},
+				"vds_lag_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Link Aggregation Group (LAG) name of Virtual Distributed Switch",
+				},
+				"vds_uplink_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Uplink name of VMware vSphere Distributed Switch (VDS)",
+				},
+			},
+		},
+	}
+}
+
+func getHostSwitchProfileIDsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Identifiers of host switch profiles to be associated with this host switch",
+		Optional:    true,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+		},
+	}
+}
+
+func getIPAssignmentSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Specification for IPs to be used with host switch virtual tunnel endpoints",
+		MaxItems:    1,
+		Required:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"assigned_by_dhcp": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Description: "Enables DHCP assignment. Should be set to true",
+				},
+				"static_ip_list": {
+					Type:        schema.TypeList,
+					MaxItems:    1,
+					Optional:    true,
+					Description: "IP assignment specification for Static IP List.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"default_gateway": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Gateway IP",
+								ValidateFunc: validateSingleIP(),
+							},
+							"ip_addresses": {
+								Type:        schema.TypeList,
+								Description: "List of IPs for transport node host switch virtual tunnel endpoints",
+								MinItems:    1,
+								Required:    true,
+								Elem: &schema.Schema{
+									Type:         schema.TypeString,
+									ValidateFunc: validateSingleIP(),
+								},
+							},
+							"subnet_mask": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Subnet mask",
+								ValidateFunc: validateSingleIP(),
+							},
+						},
+					},
+				},
+				"static_ip_mac_list": {
```

Comment:
same as above

Reply:
Same - the API allows multiple pairs here as well.